x/pairing/keeper/scores/score.go

```diff
@@ -188,30 +189,34 @@ func PickProviders(ctx sdk.Context, scores []*PairingScore, groupCount int, hash
 		return returnedProviders
 	}
 
-	for it := 0; it < groupCount; it++ {
-		hash := tendermintcrypto.Sha256(hashData) // TODO: we use cheaper algo for speed
-		bigIntHash := new(big.Int).SetBytes(hash)
-		uintHash := sdk.NewUintFromBigInt(bigIntHash)
-		modRes := uintHash.Mod(scoreSum)
+	sum256 := sha256.Sum256(hashData)
+	// Fold the SHA-256 hash into a 64-bit seed using bitwise XOR
+	var seed int64
+	for i := 0; i < len(sum256)/8; i++ {
+		seed ^= int64(binary.BigEndian.Uint64(sum256[i*8 : (i+1)*8]))
+	}
 
+	rand.Seed(seed)
```

Comment:
This should go into a helper function/submodule in e.g. `utils/rand`, e.g. something like:
```
    New(seed) *math.Rand  // returns a new RPNG instance properly seeded
    Seed(*math.Rand, seed) *math.Rand  // re-seeds an existing RPNG instance
```

And it should be accompanied with tests to verify uniform distribution, e.g. can be copied from here:
https://qqq.ninja/blog/post/fast-threadsafe-randomness-in-go/#uniform-distribution-test

Reply:
fixed 7944693a